scapy/sendrecv.py

```diff
@@ -934,7 +934,7 @@ def _run(self,
         # Get select information from the sockets
         _main_socket = next(iter(sniff_sockets))
         select_func = _main_socket.select
-        _backup_read_func = _main_socket.__class__.recv
+        read_func = _main_socket.__class__.recv
```

Comment:
Please correct me if I'm wrong, but couldn't you entirely remove the `read_func` from this function, if you later in line 985 call `p = s.recv()`